### ISSUE TITLE
Type error when accessing SQLAlchemy model's properties

### ISSUE THREAD

**Issue by shreewatsa** [microsoft/pyright#3337]
**Describe the bug**
Pyright gives type error in neovim diagnostics when trying to access values of sqlalchemy model properties. More details is in the attachment.

**To Reproduce**
Use 'pyright' as the python lsp server in neovim. Use sqlalchemy model and try to access it's properties.

**Expected behavior**
Pyright should not consider this situation as a type error.

**Screenshots or Code**
```python
from sqlalchemy import Column, Integer
from sqlalchemy.orm import declarative_base
Base = declarative_base()
class Person(Base):
    __tablename__ = "person"
    id = Column(Integer, primary_key=True)
person = Person(id=23)
uid: int = person.id
```

**VS Code extension or command-line**
Running pyright in neovim (in terminal) as python lsp server. neovim 0.6.1 ,  pyright 1.1.237 .
Not checked in vscode.

**Additional context**
What should I do fix/supress this error ? Is there any solutions/patches already available ?
![2022-04-13-135017_1920x1080_scrot](https://user-images.githubusercontent.com/20319741/163132965-6b53f3ec-f40e-479c-9ccd-b1e6e22607c7.png)

**Comment by erictraut** [comment]
Pyright is doing the right thing here from a type checking perspective. The type of `id` is correctly evaluated as `Column`, which isn't type compatible with `int`. 

You'll need to work with someone familiar with sqlalchemy and its type stubs if you think that the behavior is incorrect. The type stubs for sqlalchemy come from typeshed, so you could post an issue or a question in the typeshed github repo.

**Comment by shreewatsa** [comment]
> Which `sqlalchemy` stubs are you using?

Appreciate your query.
I wasnot using any  sqlalchemy stubs at the time of asking above. After you asked about this, I researched and installed typeshed (types-sqlalchemy) and dropbox/sqlalchemy-stubs as these seemed to solve the problem, and it worked for this particular issue. 
However, it's not working on other parts like:
![2022-04-13-221853_880x132_scrot](https://user-images.githubusercontent.com/20319741/163228590-009c7da5-5f54-48da-8ea0-769acd6585d6.png)
Here, 'Device' is a sqlalchemy model with some properties. The code works perfectly despite the type errors. I think these plugins works for mypy but not perfectly with pyright. I researched if anyone got it already but could not find much to solve it. What would you recommend to fix this issue ?

**Comment by bryanforbes** [comment]
> I researched and installed typeshed (types-sqlalchemy) and dropbox/sqlalchemy-stubs

You probably only need `dropbox/sqlalchemy-stubs`

> I researched if anyone got it already but could not find much to solve it. What would you recommend to fix this issue ?

Until SQLAlchemy 2.0 comes out, you'll just need to use a `# type: ignore` on lines like this.

**Comment by shreewatsa** [comment]
@erictraut I got your point. Really appreciate your help. Thank you.

**Comment by shreewatsa** [comment]
@bryanforbes seems like not much options. Thanks you very much.
